avro/src/main/scala/magnolify/avro/AvroType.scala

```diff
@@ -43,12 +43,11 @@ sealed trait AvroType[T] extends Converter[T, GenericRecord, GenericRecord] {
 }
 
 object AvroType {
-  implicit def apply[T: AvroField.Record]: AvroType[T] = apply(identity: CaseMapper)
+  implicit def apply[T: AvroField.Record]: AvroType[T] = AvroType(identity: CaseMapper)
 
-  def apply[T: AvroField.Record](g: CaseMapper): AvroType[T] =
+  def apply[T](g: CaseMapper)(implicit f: AvroField.Record[T]): AvroType[T] =
```

Comment:
Explicit `(implicit f)` so that we don't have to do `val f = implicitly[T]` inside.